authentication/admin.py

```diff
@@ -1,3 +1,73 @@
 from django.contrib import admin
+from django import forms
+from django.contrib.auth.admin import UserAdmin as BaseUserAdmin
+from django.contrib.auth.forms import ReadOnlyPasswordHashField
+from .email import send_welcome_email
 
-# Register your models here.
+from .models import User
+
+
+
+class AddUserForm(forms.ModelForm):
+
+    password1 = forms.CharField(label='Password', widget=forms.PasswordInput)
+    password2 = forms.CharField(label='confirm password', widget=forms.PasswordInput)
+
```

Comment:
Why are you using forms here?

Reply:
i was supposed to remove them since you said the users can register by themselves